packages/datepickers/src/types/index.ts

```diff
@@ -0,0 +1,138 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import { HTMLAttributes } from 'react';
+import { Modifiers } from 'popper.js';
+
+const SHARED_PLACEMENT = [
+  'auto',
+  'top',
+  'top-start',
+  'top-end',
+  'bottom',
+  'bottom-start',
+  'bottom-end'
+] as const;
+
+export const PLACEMENT = [
+  ...SHARED_PLACEMENT,
+  'end',
+  'end-top',
+  'end-bottom',
+  'start',
+  'start-top',
+  'start-bottom'
+] as const;
+
+export const POPPER_PLACEMENT = [
+  ...SHARED_PLACEMENT,
+  'right',
+  'right-start',
+  'right-end',
+  'left',
+  'left-start',
+  'left-end'
+] as const;
+
+export type GardenPlacement = typeof PLACEMENT[number];
```

Comment:
@jzempel could you help me understand what the bracket notation here means? Should I read it as "string lookup on the `PLACEMENT` object"? But I see that `PLACEMENT` is an array, so maybe it's an index lookup on `PLACEMENT`? I'm still soaking up TS and a bit confused 🙏🏽 .

Reply:
this SO & comments has just about as good of an explanation as any I've found: https://stackoverflow.com/a/59857409